x-pack/plugin/deprecation/qa/rest/src/javaRestTest/java/org/elasticsearch/xpack/deprecation/DeprecationHttpIT.java

```diff
@@ -230,16 +225,13 @@ public void testUniqueDeprecationResponsesMergedTogether() throws IOException {
         assertOK(response);
 
         final List<String> deprecatedWarnings = getWarningHeaders(response.getHeaders());
-        final List<Matcher<String>> headerMatchers = new ArrayList<>();
+        final List<Matcher<? super String>> headerMatchers = new ArrayList<>();
```

Comment:
String is final, so this should not be necessary?

Reply:
The parameters on `containsInAnyOrder` needs this to be `? super T`. If it was declared as `containsInAnyOrder(Collection<? extends Matcher<? super T>>)` then it would be ok, but it's `containsInAnyOrder(Collection<Matcher<? super T>>)` specifically. Go figure.